src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -518,9 +518,9 @@ public ConnectorType ConnectorType
             }
         }
 
-        private ObservableCollection<string> recentFiles =
-            new ObservableCollection<string>();
-        public ObservableCollection<string> RecentFiles
+        private SmartObservableCollection<string> recentFiles =
+            new SmartObservableCollection<string>();
+        public SmartObservableCollection<string> RecentFiles
```

Comment:
API break here, even though it is a subclass.

Reply:
I will probably remove it, since it was not really the main performance contributor here, and recent files size is anyway not going to be a huge problem. Switching it back did not make much difference.